File: bot-api/src/main/java/gg/mineral/bot/api/entity/living/player/ClientPlayerMP.java
package gg.mineral.bot.api.entity.living.player;

public interface ClientPlayerMP extends ClientPlayer {

    /**
     * Gets the player's last reported X position.
     * 
     * @return the player's last reported X position
     */
    double getLastReportedX();

    /**
     * Gets the player's last reported Y position.
     * 
     * @return the player's last reported Y position
     */
    double getLastReportedY();

    /**
     * Gets the player's last reported Z position.
     * 
     * @return the player's last reported Z position
     */
    double getLastReportedZ();

}

File: bot-api/src/main/kotlin/gg/mineral/bot/api/entity/living/player/ClientPlayerMP.kt
package gg.mineral.bot.api.entity.living.player

interface ClientPlayerMP : ClientPlayer {
    /**
     * Gets the player's last reported X position.
     *
     * @return the player's last reported X position
     */
    val lastReportedX: Double

    /**
     * Gets the player's last reported Y position.
     *
     * @return the player's last reported Y position
     */
    val lastReportedY: Double

    /**
     * Gets the player's last reported Z position.
     *
     * @return the player's last reported Z position
     */
    val lastReportedZ: Double
}
